litestar/cli/commands/core.py

```diff
@@ -319,7 +320,38 @@ def routes_command(app: Litestar, exclude: tuple[str, ...], schema: bool) -> Non
     if exclude is not None:
         sorted_routes = remove_routes_with_patterns(sorted_routes, exclude)
 
-    console.print(_RouteTree(sorted_routes))
+    if output_format == "json":
+        import inspect
+        import json
+
+        routes_list = []
+        for route in sorted_routes:
+            route_info: dict[str, Any] = {
+                "path": route.path,
+                "type": route.__class__.__name__,
+                "methods": sorted(getattr(route, "http_methods", [])),
+            }
+
+            if isinstance(route, HTTPRoute):  # pragma: no cover
+                route_info["handlers"] = [
+                    {
+                        "name": h.name,
+                        "path": h.paths,
+                        "handler": h.handler_name,
+                        "methods": sorted(h.http_methods),
+                        "async": inspect.iscoroutinefunction(unwrap_partial(h.fn)),
+                    }
+                    for h in route.route_handlers
+                ]
```

Comment:
This duplicates the logic already present in the `_RouteTree` class. I suggest extracting that logic into a common utility, which then both the tree output as well as the JSON output can use.